pkg/controller/toolchainstatus/toolchainstatus_controller_test.go

```diff
@@ -77,8 +77,8 @@ func prepareReconcile(t *testing.T, requestName string, httpTestClient *fakeHTTP
 	return r, reconcile.Request{NamespacedName: test.NamespacedName(test.HostOperatorNs, requestName)}, fakeClient
 }
 
-func prepareReconcileWithStatusConditions(t *testing.T, requestName string, conditions []toolchainv1alpha1.Condition, initObjs ...runtime.Object) (*ReconcileToolchainStatus, reconcile.Request, *test.FakeClient) {
-	reconciler, req, fakeClient := prepareReconcile(t, requestName, newResponseGood(), newGetMemberClustersFuncReady, initObjs...)
+func prepareReconcileWithStatusConditions(t *testing.T, requestName string, getMemberClustersFunc getMemberClustersFunc, conditions []toolchainv1alpha1.Condition, initObjs ...runtime.Object) (*ReconcileToolchainStatus, reconcile.Request, *test.FakeClient) {
```

Comment:
May be worth renaming either the variable or the type `getMemberClustersFunc` for readability

Reply:
fair point, I'll see what I can do about it ;)

Reply:
done in d37a2cd: code should be much simpler now:no more func with 3 levels of nested returns! 😅